File: app/src/main/java/com/seletiva/santander/investment/ui/investments/domain/Screen.java
package com.seletiva.santander.investment.ui.investments.domain;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class Screen {
    @SerializedName("title")
    private String title;

    @SerializedName("fundName")
    private String fundName;

    @SerializedName("whatIs")
    private String whatIs;

    @SerializedName("definition")
    private String definition;

    @SerializedName("riskTitle")
    private String riskTitle;

    @SerializedName("risk")
    private int risk;

    @SerializedName("infoTitle")
    private String infoTitle;

    @SerializedName("moreInfo")
    private MoreInfo moreInfo;

    @SerializedName("info")
    private List<InfoItem> info;


    @SerializedName("downInfo")
    private List<InfoItem> downInfo;

    public List<InfoItem> getAllInfoItems() {
        List<InfoItem> allItems = new ArrayList<>();

        if (info != null) {
            allItems.addAll(info);
        }

        if (downInfo != null) {
            allItems.addAll(downInfo);
        }

        return allItems;
    }

    public String getTitle() {
        return title;
    }

    public String getFundName() {
        return fundName;
    }

    public String getWhatIs() {
        return whatIs;
    }

    public String getDefinition() {
        return definition;
    }

    public String getRiskTitle() {
        return riskTitle;
    }

    public int getRisk() {
        return risk;
    }

    public String getInfoTitle() {
        return infoTitle;
    }

    public MoreInfo getMoreInfo() {
        return moreInfo;
    }

    public List<InfoItem> getInfo() {
        return info;
    }

    public List<InfoItem> getDownInfo() {
        return downInfo;
    }
}

File: SantanderTest/app/src/main/java/lucasonofre/santandertest/model/Screen.kt
package lucasonofre.santandertest.model

import com.google.gson.annotations.SerializedName

/**
 * Classe modelo do item da chamada
 */
class Screen{
    @SerializedName("screen")
    var screen: ScreenItens? = null
}

/**
 * Classe modelo de cada item da chamada
 */
 class ScreenItens {

     @SerializedName("title")
     val title:  String? = null

     @SerializedName("fundName")
     val fundName: String? = null

     @SerializedName("whatIs")
     val whatIs: String? = null

     @SerializedName("definition")
     val definition: String? = null

     @SerializedName("riskTitle")
     val riskTitle: String? = null

     @SerializedName("risk")
     val risk: Long? = null

     @SerializedName("infoTitle")
     val infoTitle: String? = null

     @SerializedName("moreInfo")
     val moreInfo: MoreInfo? = null

     @SerializedName("info")
     val info: ArrayList<Info>? = null

     @SerializedName("downInfo")
     val downInfo: ArrayList<DownInfo>? = null
 }

 class MoreInfo {
     @SerializedName("month")
     val month: Yield? = null

     @SerializedName("year")
     val year: Yield? = null

     @SerializedName("12months")
     val the12Months: Yield? = null
 }

/**
 * Classe modelo usada para representar os lucros
 */

 class Yield {

    @SerializedName("fund")
    var fund: Float? = null

    @SerializedName("CDI")
    var cdi: Float? = null

}

class Info {
    @SerializedName("name")
    val name: String? = null

    @SerializedName("data")
    val data: String? = null

}

class DownInfo {
    @SerializedName("name")
    val name: String? = null

    @SerializedName("data")
    val data: String? = null

}

class YieldListItem{

    var yield:Yield?      = null
    var titleItem:String? = null

    constructor(yield: Yield?, titleItem:String?) {
        this.yield     = `yield`
        this.titleItem = titleItem

    }
}